pkg/sidecar/sidecar.go

```diff
@@ -21,18 +21,43 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+// Whenever an attempt is made to signal pid_file_name, the outcome is sent
+// in messages on a channel with this type. Mainly for test purposes.
+type pidFileSignalledResult struct {
+	pid int
+	err error
+}
+
 // Sidecar is the component that consumes the Workload API and renews certs
-// implements the interface Sidecar
 type Sidecar struct {
 	config         *Config
 	client         *workloadapi.Client
 	jwtSource      *workloadapi.JWTSource
 	processRunning bool
 	process        *os.Process
-	certReadyChan  chan struct{}
-	health         Health
 
+	// Mutex to protect processRunning
 	mu sync.Mutex
+
+	// Health server
+	health Health
+
+	// When a new x.509 SVID is received, it is sent to this channel. Mainly
+	// for test purposes. Do not close.
+	certReadyChan chan *workloadapi.X509Context
+
+	// When 'cmd' exits and wait() returns, the exit status is sent to this
+	// channel. Mainly for test purposes. Do not close.
+	cmdExitChan chan os.ProcessState
+
+	// When the process is signaled to reload certificates the outcome is
+	// sent to this channel. Mainly for test purposes. Do not close.
+	pidFileSignalledChan chan pidFileSignalledResult
+
+	// stdio to connect to the 'cmd' to run
+	stdin  *os.File
+	stdout *os.File
+	stderr *os.File
```

Comment:
What is this being used for?

Reply:
So stdio to/from the sidecar can be captured and redirected in tests.

There are places that rely on creating files that should probably just read from the workload being run instead.

I'm not sure I got around to adding cases that use them though. Could remove if preferred.

Reply:
Yes, please remove if not being used.

Reply:
OK, went back and checked since I'd forgotten the details. 

The intent was that stdout and stderr are assigned to nil in some tests to suppress unwanted output. See e.g. pkg/sidecar/sidecar_test.go:111 and :128 . It looks like I made an error in doing so, where I failed to differentiate between "nil because it wasn't set" and "nil because it was explicitly set to nil".

The effect was supressing stdio for all the tests. Which was harmless since none of them use stdio, but not the intended effect.

I've pushed a fix. It now passes stdin/out/err by default, but supports capturing and logging it only in `go test -v` for case where "normal" errors from a command run should be suppressed.